lib/matplotlib/tests/test_collections.py

```diff
@@ -747,3 +749,71 @@ def test_legend_inverse_size_label_relationship():
     handle_sizes = [5 / x**2 for x in handle_sizes]
 
     assert_array_almost_equal(handle_sizes, legend_sizes, decimal=1)
+
+
+@pytest.mark.style('default')
+@pytest.mark.parametrize('pcfunc', [plt.pcolor, plt.pcolormesh])
+def test_color_logic(pcfunc):
+    rgba_none = mcolors.to_rgba_array('none')
+    z = np.arange(12).reshape(3, 4)
+    pc = pcfunc(z, edgecolors='red', facecolors='none')
+    face_default = mcolors.to_rgba_array(pc._get_default_facecolor())
+    assert_array_equal(pc.get_edgecolor(), [[1, 0, 0, 1]])
+    # Check setting attributes after initialization:
+    pc = pcfunc(z)
+    pc.set_facecolor('none')
+    pc.set_edgecolor('red')
+    assert_array_equal(pc.get_edgecolor(), [[1, 0, 0, 1]])
+    pc.set_alpha(0.5)
+    assert_array_equal(pc.get_edgecolor(), [[1, 0, 0, 0.5]])
+    pc.set_edgecolor(None)  # reset to default
+    pc.set_alpha(None)  # restore default alpha
+    pc.update_scalarmappable()
+    assert pc.get_edgecolor().shape == (12, 4)  # color-mapped
+    pc.set_facecolor(None)
+    pc.update_scalarmappable()
+    assert pc.get_facecolor().shape == (12, 4)  # color-mapped
+    assert_array_equal(pc.get_edgecolor(), rgba_none)  # default: 'none'
```

Comment:
For this, can you use `mcolors.same_color` (I _think_ it does the broadcast correctly)?
```suggestion
    assert mcolors.same_color(pc.get_edgecolor(), 'none')  # default: 'none'
```

Reply:
This can be done, but I think it makes the test less useful.  If there is a failure, pytest won't be able to give any additional information: what was the expected result, and what was the actual result.

Reply:
No, default pytest does exception re-writing and will print out all arguments in a failed assert:
```pytb
__________ test_color_logic[pcolormesh] ____________

pcfunc = <function pcolormesh at 0x7f2ed630eb00>

    @pytest.mark.style('default')
    @pytest.mark.parametrize('pcfunc', [plt.pcolor, plt.pcolormesh])
    def test_color_logic(pcfunc):
        rgba_none = mcolors.to_rgba_array('none')
        z = np.arange(12).reshape(3, 4)
        pc = pcfunc(z, edgecolors='red', facecolors='none')
        face_default = mcolors.to_rgba_array(pc._get_default_facecolor())
        assert_array_equal(pc.get_edgecolor(), [[1, 0, 0, 1]])
        # Check setting attributes after initialization:
        pc = pcfunc(z)
        pc.set_facecolor('none')
        pc.set_edgecolor('red')
        assert_array_equal(pc.get_edgecolor(), [[1, 0, 0, 1]])
        pc.set_alpha(0.5)
        assert_array_equal(pc.get_edgecolor(), [[1, 0, 0, 0.5]])
        pc.set_edgecolor(None)  # reset to default
        pc.set_alpha(None)  # restore default alpha
        pc.update_scalarmappable()
        assert pc.get_edgecolor().shape == (12, 4)  # color-mapped
        pc.set_facecolor(None)
        pc.update_scalarmappable()
        assert pc.get_facecolor().shape == (12, 4)  # color-mapped
>       assert mcolors.same_color(pc.get_edgecolor(), 'b')  # instead of 'none'
E       AssertionError: assert False
E        +  where False = <function same_color at 0x7f2ed7407170>(array([], shape=(0, 4), dtype=float64), 'b')
E        +    where <function same_color at 0x7f2ed7407170> = mcolors.same_color
E        +    and   array([], shape=(0, 4), dtype=float64) = <bound method Collection.get_edgecolor of <matplotlib.collections.QuadMesh object at 0x7f2ed2171e90>>()
E        +      where <bound method Collection.get_edgecolor of <matplotlib.collections.QuadMesh object at 0x7f2ed2171e90>> = <matplotlib.collections.QuadMesh object at 0x7f2ed2171e90>.get_edgecolor

lib/matplotlib/tests/test_collections.py:776: AssertionError
```